paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java

```diff
@@ -230,14 +231,21 @@ public static DataStream<RowData> buildSource(
             ReadBuilder readBuilder,
             long monitorInterval,
             boolean emitSnapshotWatermark) {
+        KeySelector<Split, Integer> keySelector =
+                split -> {
+                    if (((DataSplit) split).partition() != null) {
```

Comment:
You can just use `ChannelComputer.select`.

Reply:
Updated